composites/SnippetPreview/components/Button.js

```diff
@@ -62,3 +68,50 @@ export const SnippetPreviewButton = styled( BaseButton )`
 	line-height: 15px;
 	font-size: 0.8rem;
 `;
+
+/**
+ * Applies styles to icon for IconButton with text.
+ *
+ * @param {ReactElement} icon The original icon.
+ *
+ * @returns {ReactElement} Icon with text styles applied.
+ */
+function applyIconTextStyle( icon ) {
+	return styled( icon )`
+		margin: 0 8px 0 0;
+		float: left;
+	`;
+}
+
+/**
+ * Returns an icon button that can optionally contain text.
+ *
+ * @param {object} props Component props.
+ *
+ * @returns {ReactElement} Styled icon button.
+ */
+export const IconButton = ( props ) => {
+	const { children, icon, iconColor } = props;
+
+	let IconComponent = Icon;
+	if( children ) {
+		IconComponent = applyIconTextStyle( IconComponent );
+	}
+
+	return (
+		<SnippetPreviewButton { ...props } >
+			<IconComponent icon={ icon } color={ iconColor } />
+			{ children }
+		</SnippetPreviewButton>
+	);
+};
+
+IconButton.propTypes = {
+	icon: PropTypes.func.isRequired,
+	iconColor: PropTypes.string,
+	children: PropTypes.string,
+};
+
+IconButton.defaultProps = {
+	icon: "edit",
```

Comment:
This shouldn't be a default (and you've already made this prop required, so this default doesn't do anything)